src/peer/mod.rs

```diff
@@ -111,14 +111,14 @@ where
     let con = Connection::connect(relay, twin, signer.clone());
     let mut address = Address::new();
     address.twin = twin;
-
+    let upstream_db = db.clone();
```

Comment:
instead of declaring a new variable you can change the call to Downstream::new to `db.clone()` then pass the db to the Upstream::new 🤷🏼‍♂️